### ISSUE TITLE
Within K0 category clustering

### ISSUE THREAD

**Issue by kdiverson** [SchlossLab/opf#37]

**Comment by kdiverson** [comment]
If this is how we want to do OPFs by default, clustering within category even for small BLAST files, I think it would be a good idea to break up the genes by K0 before doing the all v all BLASTs and then running 'K0 genes v all' BLASTs for all K0s individually. Doing this is technically the same as a 'regular' all v all BLAST because the databases are the same size so things like e-value and BLAST scores aren't altered, right @pschloss ? The way I've been doing it now is BLASTing all the genes and then parsing the big BLAST file.

**Comment by pschloss** [comment]
Correct. This is what we do with cluster.split for 16S rRNA genes. There we split by taxonomy and then cluster within each taxonomic group. The risk there is that if we split too finely (e.g. to genus) and then cluster too broadly (e.g. distance of 0.10) then we have multiple OTUs that should really be the same one but cluster separately because they're from different genera.

I suspect the risk for that is much less here. It would be great to split to KO and then cluster within those. Of course, the unclassified/unknown will still be the largest, but it should cut down on the all-v-all considerably. I think we didn't do that before because we wanted to show that all-v-all and split and cluster gave similar results. This is still an important result to show, but going forward, there's no need to do the full all-v-all.

**Comment by kdiverson** [comment]
Thanks, I'll update the documentation and code to reflect this.
